web/src/components/MapView.tsx

```diff
@@ -47,18 +51,18 @@ const geolocateControlStyle: React.CSSProperties = {
 
 type MapViewProps = {
   bboxViewport: MapViewViewport
-  featureCollection: PoiFeatureCollection
+  featureCollection: PoiFeatureCollection | null
+  currentFeature: PoiFeature | null
+  setCurrentFeature: (feature: PoiFeature | null) => void
 }
 
-const MapView: React.FunctionComponent<MapViewProps> = (props: MapViewProps): ReactElement => {
-  const { featureCollection, bboxViewport } = props
+const MapView = React.forwardRef((props: MapViewProps, ref: React.Ref<BottomSheetRef>): ReactElement => {
```

Comment:
Why do we need these kind of `forwardRef` components? It should be enough to pass the ref as regular prop. Or even better: Can't we just pass a `onTouchMove` function prop and `setCurrentFeature` as
``` ts
  selectFeature = (feature: PoiFeature | null) => {
    setCurrentFeature(feature)
    // set snap point as desired
  }
```
This component should not have to handle snap points.

Reply:
very good hint. thx